### PR TITLE
Add delete functionality

### PR DESCRIPTION
**Pull request by Leboroz**
# Add delete functionality

## Features added:
- I added new JavaScript files and import them as a module:
  - it will contain methods related to the status updates.
- I added an event listener to the checkboxes.
- I updated the item's object value for the completed key upon user actions.
- I Implemented a function for the "Clear all completed" button.
- I stored the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: src/components/todoitem.js [hunk 1]
```diff
@@ -37,10 +39,7 @@ export default class TodoItem {
     }
     check.addEventListener('change', (e) => {
       this.completed = !this.completed;
-      //* if checkbox is checked input is disabled
-      if (e.target.checked) input.setAttribute('disabled', 'true');
-      //* if checkbox is not checked input is enabled
-      else if (!e.target.checked) input.removeAttribute('disabled');
+      checkHandler(e, input);
```
Comment (descholar-ceo) [inline src/components/todoitem.js]: - [x] Upon marking a given todo item as _completed_ , the cahnges are not being reflected in local storage hence on page reloading, the changes disappear, would you fix that issue?
- [x] The same issue happens when we edit a given todo item, kindly watch the following screencast for more context


https://user-images.githubusercontent.com/56886064/149422443-e2a14af1-138b-470f-b84f-356e2d546d8d.mp4

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @Leboroz , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/todoitem.js (hunk 1 above)
Comment (Leboroz) [reply src/components/todoitem.js]: Thank you for the video 👍

**Comment by omar-labana** [comment]
Hi @Leboroz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**